torch/_inductor/codegen/cpp.py

```diff
@@ -1231,6 +1231,30 @@ def bitwise_left_shift(a, b):
     def bitwise_right_shift(a, b):
         return f"{a} >> {b}"
 
+    @staticmethod
+    def remainder(a, b):
+        # Since we register remainder as ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT in
+        # the lowering phase: https://github.com/pytorch/pytorch/blob/
+        # eb1583dbc1a8c4a8f0d1525438f16350f11f1208/torch/_inductor/lowering.py#L5853
+        # Should expect same dtype for a and b. Otherwise, we need to handle
+        # the case when a and b with mixed integer and float.
+        assert (
+            a.dtype == b.dtype
+        ), "remainder vec implementation expect the same inputs' dtype."
+        if is_float_dtype(a.dtype):
+            return f"{a} - ({a} / {b}).floor() * {b}"
+        else:
+            assert is_integer_dtype(a.dtype)
+            if a.dtype in [torch.uint8, torch.int8]:
```

Comment:
how about int16?

Reply:
I think int16 has not been support generally in vectorization yet as 
https://github.com/pytorch/pytorch/blob/d0ad13fa42fc2e9935bd3bda2937a3491276d274/torch/_inductor/codegen/cpp.py#L2819-L2828
I think we may need another PR to support int16 generally.